internal/controller/clusterdeployment_controller.go

```diff
@@ -388,13 +389,54 @@ func (r *ClusterDeploymentReconciler) updateCluster(ctx context.Context, mc *kcm
 	return ctrl.Result{}, nil
 }
 
+func (r *ClusterDeploymentReconciler) updateSveltosClusterCondition(ctx context.Context, clusterDeployment *kcm.ClusterDeployment) (bool, error) {
+	sveltosClusters := &libsveltosv1beta1.SveltosClusterList{}
+
+	err := r.Client.List(ctx, sveltosClusters, &client.ListOptions{
+		LabelSelector: labels.SelectorFromSet(map[string]string{kcm.FluxHelmChartNameKey: clusterDeployment.Name}),
+	})
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			return true, fmt.Errorf("failed to get sveltos cluster status: %w", err)
+		}
+	}
```

Comment:
```suggestion
	if err := r.Client.List(ctx, sveltosClusters, &client.ListOptions{
		LabelSelector: labels.SelectorFromSet(map[string]string{kcm.FluxHelmChartNameKey: clusterDeployment.Name}),
	}); err != nil {
		return true, fmt.Errorf("failed to get sveltos cluster status: %w", err)
	}
```

redundant check, the instance itself is not null, the list operation does not return such an error basically (until the resource itself is missing which won't arise)